File: backend/app/models/bom.py
from pydantic import BaseModel
from typing import Optional


class BomItem(BaseModel):
    id: str
    category: str       # 분야 (IT 장비, 기반시설)
    subcategory: str    # 항목 (Server, Storage, Network, ...)
    equipment: str      # Equipment 이름
    quantity: int
    description: Optional[str] = ""
    unit_power_w: Optional[float] = None   # W
    unit_height_u: Optional[int] = None    # U


class BomImportResult(BaseModel):
    items: list[BomItem]
    source: str  # "pptx" | "excel" | "csv"
    warnings: list[str] = []

File: frontend/src/api/bom.ts
import axios from 'axios'

const api = axios.create({ baseURL: '' })

export interface BomItem {
  id: string
  category: string
  subcategory: string
  equipment: string
  quantity: number
  description: string
  unit_power_w: number | null
  unit_height_u: number | null
}

export interface BomImportResult {
  items: BomItem[]
  source: string
  warnings: string[]
}

export const importBom = async (file: File): Promise<BomImportResult> => {
  const form = new FormData()
  form.append('file', file)
  const res = await api.post<BomImportResult>('/api/bom/import', form)
  return res.data
}

export const getSampleBom = async (): Promise<BomImportResult> => {
  const res = await api.get<BomImportResult>('/api/bom/sample')
  return res.data
}
